### PR TITLE
Search functionality

### PR DESCRIPTION
**Pull request by cdalt**
added food path variable. allows for unique searching for recipe ingredients.

### CODE REVIEW

File: src/main/kotlin/com/example/Food_API/controller/FoodController.kt [hunk 1]
```diff
@@ -7,18 +7,19 @@ import org.springframework.web.bind.annotation.*
 @RestController
 @RequestMapping("/api/food")
 class FoodController() {
-    @GetMapping(produces = ["application/json"])
-    fun getRecipes(): String? {
+
+    @GetMapping("/{foodId}")
```
Comment (Rufio0425) [inline src/main/kotlin/com/example/Food_API/controller/FoodController.kt:11]: For the `@GetMapping()` you can still add the `produces`. It would look something like `@GetMapping("/{foodId}", produces = ["application/json"])`

File: src/main/kotlin/com/example/Food_API/controller/FoodController.kt [hunk 2]
```diff
@@ -7,18 +7,19 @@ import org.springframework.web.bind.annotation.*
...
+    fun findFoodById(@PathVariable foodId: Int): String? {
```
Comment (Rufio0425) [inline src/main/kotlin/com/example/Food_API/controller/FoodController.kt:12]: I would suggest moving the logic for this method into some sort of service interface, maybe a `FoodService` interface? That way if you do add more functions to this controller (which I'm sure you will) the controller will be easier to read and more cleaned up. This also doesn't have to be done now, but should be addressed in another PR.

File: src/main/kotlin/com/example/Food_API/models/Food.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.example.Food_API.models
```
Comment (Rufio0425) [inline src/main/kotlin/com/example/Food_API/models/Food.kt:1]: You may not need this `Food.kt` data class, unless you're assuming to eventually convert the response body from the spoonacular API to this data type and return that to the consumer of your API.

**Review by Rufio0425 — COMMENTED** [review]
Left a few comments, I think in general it looks good! Maybe remove the Food data class if you're not using it right now, and defining the `produces` for the GetMapping on `findFoodById()`

File: src/main/kotlin/com/example/Food_API/controller/FoodController.kt (hunk 2 above)
Comment (cdalt) [reply src/main/kotlin/com/example/Food_API/controller/FoodController.kt:12]: sounds good! That was going to be my next step was to add the logic to the service interface.

Comment (Rufio0425) [reply src/main/kotlin/com/example/Food_API/controller/FoodController.kt:12]: 👍

File: src/main/kotlin/com/example/Food_API/models/Food.kt (hunk 3 above)
Comment (cdalt) [reply src/main/kotlin/com/example/Food_API/models/Food.kt:1]: I removed the food class, and added the produces to get the getmapping, in a different branch and i'll be opening a PR for those changes soon.
